src/Agent/NewRelic/Agent/Core/Configuration/ReportedConfiguration.cs

```diff
@@ -1,114 +1,607 @@
 // Copyright 2020 New Relic, Inc. All rights reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+using System;
 using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using NewRelic.Agent.Configuration;
 using Newtonsoft.Json;
 
 namespace NewRelic.Agent.Core.Configuration
 {
     /// <summary>
-    /// The configuration that is reported to the collector using the agent_settings command.
+    /// The configuration that is reported to the collector using the agent_settings command, and in connect.settings.
     /// </summary>
-    public class ReportedConfiguration
+    [JsonObject(ItemNullValueHandling = NullValueHandling.Ignore)]
+    public class ReportedConfiguration : IConfiguration
     {
-        [JsonProperty("agent")]
+        private readonly IConfiguration _configuration;
+        public ReportedConfiguration(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
+        //[JsonProperty("agent.name")]
+        [JsonProperty("name")]
```

Comment:
So, this value (".NET Agent") was being reported under the key name "agent" and will now be reported under key name "name".  Is there any concern that this change might break anything?

I don't see anything in BMDS that looks like a problem.